packages/cli/generators/repository/index.js

```diff
@@ -38,6 +53,34 @@ module.exports = class RepositoryGenerator extends ArtifactGenerator {
     super(args, opts);
   }
 
+  /**
+  * Parses the files of the target directory and returns matching filename for
+  * the artifact requested
+  * @param {string} dir The target directory from which to load artifacts.
+  * @param {string} artifactName The artifact name (ex. "TodoRepository")
+  * @param {string} artifactType The artifact type (ex. "model", "repository")
+  */
+  async _findArtifactPaths(dir, artifactName, artifactType) {
+    const paths = await utils.findArtifactPaths(dir, artifactType);
+    debug(`Repository artifact paths: ${paths}`);
+
+    //Remove .ts from the paths
+    const partialPaths = paths.map(p => {
+      return path.parse(_.last(_.split(p, path.sep))).name;
+    });
+
+    //Search for the filename
+    const artifactPaths = _.filter(partialPaths, function (p) {
+      const firstWord = pascalCase(utils.toClassName(_.first(_.split(_.last
+        (_.split(p, path.sep)), '.'))));
+      const fullName = utils.toClassName(firstWord) +
+        utils.toClassName(artifactType);
+      return _.eq(artifactName, fullName);
+    });
+
+    return _.first(artifactPaths);
+  }
```

Comment:
This can be possibly simplified by using `utils` to map the artifact name to a file name instead of inferring the artifact name from the file names.

Reply:
You mean using AST? Actually, all the CLIs use the same logic... take the first word of the filename as the artifact name...

Reply:
No. I meant to say to use a function to infer the file name from the artifact name and do the match.

Reply:
Done